zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -652,13 +652,11 @@ private void updateState() {
         retryFuture,
         (bool, throwable) -> {
           if (throwable != null) {
-            LOG.error(ERROR_MESSAGE_UPDATE_STATE_FAILED, currentRecord, metadata, throwable);
-            onError(
-                throwable,
-                () -> {
-                  errorHandlingInTransaction(throwable);
-                  updateState();
-                });
+            // Log entries were written already, but committing state changes failed with an
```

Comment:
🔧 Can you change the state machine documentation above  then as well https://github.com/camunda/camunda/pull/31459/files#diff-aca559d3c449b7f34472294c4e1de56a525bc6a80b875de0d4328380d543f4f2L83-L95

Reply:
❓ Does it make sense to switch the order then? 🤔 I think before we had it like this because it could failed to write to the dispatcher, but now it can only fail when OOD? 🤔 And it would be retried on IOE? This would also be the case for the update state?

Reply:
Hm good point, I haven't considered that at all. I think we should still stick with the current order though, first writing follow up records, then committing state. Otherwise, we might commit state, then restart. That'd be an unrecoverable scenario because state and log have diverged again. The current node has committed state changes that others will never see because there are no corresponding log entries.

Reply:
Please change the doc as above mentioned https://github.com/camunda/camunda/pull/31459/files#diff-aca559d3c449b7f34472294c4e1de56a525bc6a80b875de0d4328380d543f4f2L83-L95

Reply:
Done with https://github.com/camunda/camunda/pull/31459/commits/8034cee2cb9b0d0a6757a6ac0bb1401c217f1975